src/script/components/collection/Collection.tsx

```diff
@@ -0,0 +1,225 @@
+      .then(allMessages => setMessages(allMessages as ContentMessage[]));
+  }, []);
+
+  useEffect(() => {
+    const addItem = (message: ContentMessage) => {
+      setMessages(oldMessages => [message].concat(oldMessages));
+    };
+
+    const removeItem = (messageId: string, conversationId: string) => {
+      if (conversation.id !== conversationId) {
+        // A message from a different converation, nothing to do
+        return;
+      }
+      setMessages(oldMessages => oldMessages.filter(message => message.id !== messageId));
+    };
+
+    const removeMessage = (message: ContentMessage) => {
+      removeItem(message.id, message.conversation_id);
+    };
+
+    amplify.subscribe(WebAppEvents.CONVERSATION.EPHEMERAL_MESSAGE_TIMEOUT, removeMessage);
+    amplify.subscribe(WebAppEvents.CONVERSATION.MESSAGE.ADDED, addItem);
+    amplify.subscribe(WebAppEvents.CONVERSATION.MESSAGE.REMOVED, removeItem);
+    return () => {
+      amplify.unsubscribe(WebAppEvents.CONVERSATION.EPHEMERAL_MESSAGE_TIMEOUT, removeMessage);
+      amplify.unsubscribe(WebAppEvents.CONVERSATION.MESSAGE.ADDED, addItem);
+      amplify.unsubscribe(WebAppEvents.CONVERSATION.MESSAGE.REMOVED, removeItem);
+    };
+  }, []);
+
+  const categories = splitIntoCategories(messages);
+  const {images, audio, links, files} = categories;
+
+  if (detailCategory && categories[detailCategory].length > 0) {
+    return (
+      <CollectionDetails
+        conversation={conversation}
+        messages={categories[detailCategory]}
+        onClose={() => setDetailCategory(undefined)}
+      />
+    );
+  }
+
+  const content = searchTerm ? null : (
+    <>
+      <Section
+        messages={images}
+        limit={12}
+        uieName={'collection-section-image'}
+        onSelect={() => setDetailCategory('images')}
+      >
+        <span className={`collection-header-icon icon-library`}></span>
+        <span className="label-bold-xs">{t('collectionSectionImages')}</span>
+      </Section>
+      <Section
+        messages={links}
+        limit={4}
+        uieName={'collection-section-link'}
+        onSelect={() => setDetailCategory('links')}
+      >
+        <span className={`collection-header-icon icon-link`}></span>
+        <span className="label-bold-xs">{t('collectionSectionLinks')}</span>
+      </Section>
+      <Section
+        messages={audio}
+        limit={4}
+        uieName={'collection-section-audio'}
+        onSelect={() => setDetailCategory('audio')}
+      >
+        <Icon.MicOn className="collection-header-icon" />
+        <span className="label-bold-xs">{t('collectionSectionAudio')}</span>
+      </Section>
+      <Section
+        messages={files}
+        limit={4}
+        uieName={'collection-section-files'}
+        onSelect={() => setDetailCategory('files')}
+      >
+        <span className={`collection-header-icon icon-link`}></span>
+        <span className="label-bold-xs">{t('collectionSectionFiles')}</span>
```

Comment:
this child with the label appears in every <Section/> iteration, does it make sense to add it to the component and simply pass in the translation?

Reply:
That's what I thought, initially, but if you look closer, one on the section use a different tag for the icon. Only for this one I need to use the `children`. Not great, I know

Reply:
im a bit confused by this comment tbh, these labels are all the same